src/Composer/DependencyResolver/Problem.php

```diff
@@ -649,7 +649,7 @@ private static function getProvidersList(RepositorySet $repositorySet, string $p
         $providers = $repositorySet->getProviders($packageName);
         if (\count($providers) > 0) {
             $providersStr = implode(array_map(static function ($p): string {
-                $description = $p['description'] !== '' ? ' '.substr($p['description'] ?? '', 0, 100) : '';
+                $description = $p['description'] !== '' && $p['description'] !== null ? ' '.substr($p['description'] ?? '', 0, 100) : '';
```

Comment:
Is this version more clear to the intent? We don't need the `??` operator inside the `substr()` call any more.
```suggestion
                $description = (($p['description'] ?? '') !== '') ? ' '.substr($p['description'], 0, 100) : '';
```

Reply:
I think the condition used by @xabbuh is clearer when reading the code.
But we can indeed remove the `?? ''` in the `substr` call.